src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ObjectCreationCompletionProviderTests.cs

```diff
@@ -262,5 +262,113 @@ static void Main(string[] args)
 }";
             VerifyItemExists(markup, "Program");
         }
+
+        [WorkItem(4115, "https://github.com/dotnet/roslyn/issues/4115")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public void CommitObjectWithParenthesis1()
+        {
+            var markup = @"
+class C
+{
+    void M1()
+    {
+        object o = new $$
+    }
+}";
+
+            var expected = @"
+class C
+{
+    void M1()
+    {
+        object o = new object
+    }
+}";
+
+            VerifyProviderCommit(markup, "object", expected, '(', "");
+        }
+
+        [WorkItem(4115, "https://github.com/dotnet/roslyn/issues/4115")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public void CommitObjectWithParenthesis2()
+        {
+            var markup = @"
+class C
+{
+    void M1()
+    {
+        M2(new $$
+    }
+
+    void M2(object o) { }
+}";
+
+            var expected = @"
+class C
+{
+    void M1()
+    {
+        M2(new object
+    }
+
+    void M2(object o) { }
+}";
+
+            VerifyProviderCommit(markup, "object", expected, '(', "");
+        }
+
+        [WorkItem(4115, "https://github.com/dotnet/roslyn/issues/4115")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public void DontCommitObjectWithOpenBrace1()
+        {
+            var markup = @"
+class C
+{
+    void M1()
+    {
+        object o = new $$
+    }
+}";
+
+            var expected = @"
+class C
+{
+    void M1()
+    {
+        object o = new 
+    }
+}";
+
+            VerifyProviderCommit(markup, "object", expected, '{', "");
+        }
+
+        [WorkItem(4115, "https://github.com/dotnet/roslyn/issues/4115")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public void DontCommitObjectWithOpenBrace2()
+        {
+            var markup = @"
+class C
+{
+    void M1()
+    {
+        M2(new $$
+    }
+
+    void M2(object o) { }
+}";
+
+            var expected = @"
+class C
+{
+    void M1()
+    {
+        M2(new 
+    }
```

Comment:
If { isn't a commit chat, does the list actually dismiss, or does it stay up with nothing selected?